server/src/main/java/org/apache/druid/segment/realtime/appenderator/SinkSchemaUtil.java

```diff
@@ -0,0 +1,165 @@
+                segmentId.toString(),
+                false,
+                numRows,
+                newColumns,
+                Collections.emptyList(),
+                columnMapping
+            );
+        sinkSchemas.add(segmentSchema);
+      }
+    }
+
+    return Optional.ofNullable(sinkSchemas.isEmpty() ? null : new SegmentSchemas(sinkSchemas));
+  }
+
+  /**
+   * Compute schema change for the sinks.
+   */
+  @VisibleForTesting
+  static Optional<SegmentSchemas> computeSchemaChange(
+      Map<SegmentId, Pair<RowSignature, Integer>> previousSinkSignatureMap,
+      Map<SegmentId, Pair<RowSignature, Integer>> currentSinkSignatureMap
+  )
+  {
+    List<SegmentSchema> sinkSchemas = new ArrayList<>();
+
+    for (Map.Entry<SegmentId, Pair<RowSignature, Integer>> entry : currentSinkSignatureMap.entrySet()) {
+      SegmentId segmentId = entry.getKey();
+      RowSignature sinkSignature = entry.getValue().lhs;
+
+      Integer numRows = entry.getValue().rhs;
+
+      List<String> newColumns = new ArrayList<>();
+      List<String> updatedColumns = new ArrayList<>();
+      Map<String, ColumnType> columnMapping = new HashMap<>();
+
+      // whether there are any changes to be published
+      boolean publish = false;
+      // if the resultant schema is delta
+      boolean delta = false;
+
+      if (!previousSinkSignatureMap.containsKey(segmentId)) {
+        // new Sink
+        for (String column : sinkSignature.getColumnNames()) {
+          newColumns.add(column);
+          sinkSignature.getColumnType(column).ifPresent(type -> columnMapping.put(column, type));
+        }
+        if (newColumns.size() > 0 || numRows > 0) {
+          publish = true;
+        }
+      } else {
+        RowSignature previousSinkSignature = previousSinkSignatureMap.get(segmentId).lhs;
+        Set<String> previousSinkDimensions = new HashSet<>(previousSinkSignature.getColumnNames());
+
+        Integer previousNumRows = previousSinkSignatureMap.get(segmentId).rhs;
+        for (String column : sinkSignature.getColumnNames()) {
+          boolean added = false;
+          if (!previousSinkDimensions.contains(column)) {
+            newColumns.add(column);
+            added = true;
+          } else if (!Objects.equals(previousSinkSignature.getColumnType(column), sinkSignature.getColumnType(column))) {
+            updatedColumns.add(column);
+            added = true;
+          }
+
+          if (added) {
+            sinkSignature.getColumnType(column).ifPresent(type -> columnMapping.put(column, type));
+          }
+        }
+
+        if ((!Objects.equals(numRows, previousNumRows)) || (updatedColumns.size() > 0) || (newColumns.size() > 0)) {
+          publish = true;
+          delta = true;
+        }
+      }
+
+      if (publish) {
+        SegmentSchema segmentSchema =
+            new SegmentSchema(
+                segmentId.getDataSource(),
+                segmentId.toString(),
+                delta,
```

Comment:
```suggestion
                true,
```

Reply:
Delta can be either true or false. Hence cannot hardcode.